extensions/data-plane/data-plane-spi/src/main/java/org/eclipse/dataspaceconnector/dataplane/spi/pipeline/DataTransferExecutorServiceContainer.java

```diff
@@ -0,0 +1,33 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+package org.eclipse.dataspaceconnector.dataplane.spi.pipeline;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.concurrent.ExecutorService;
+
+/**
+ * Holder class for a shared {@link ExecutorService} across data transfer implementations.
+ */
+public class DataTransferExecutorServiceContainer {
```

Comment:
It is possible, but this ensures domain isolation. Otherwise we would for example have a service conflict if some other module were to register another ExecutorService.

Discussed with Paul in order to have service qualifiers prioritized for the next milestone.